### PR TITLE
Staging next that includes gcc8

### PR DESCRIPTION
**Pull request by FRidh**
###### Things done

- [ ] Tested using sandboxing ([nix.useSandbox](http://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](http://nixos.org/nix/manual/#sec-conf-file) on non-NixOS)
- Built on platform(s)
   - [ ] NixOS
   - [ ] macOS
   - [ ] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nix-review --run "nix-review wip"`
- [ ] Tested execution of all binary files (usually in `./result/bin/`)
- [ ] Determined the impact on package closure size (by running `nix path-info -S` before and after)
- [ ] Ensured that relevant documentation is up to date
- [ ] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

###### Notify maintainers

cc @

### CODE REVIEW

**Comment by FRidh** [comment]
elfutils failure for aarach64 https://hydra.nixos.org/build/99978054

**Comment by jonringer** [comment]
@GrahamcOfBorg eval

**Comment by FRidh** [comment]
@vcunat can you cancel the staging-next builds of all but the last evaluation?

**Comment by vcunat** [comment]
OK, I also triggered an evaluation, as I can see ~5k rebuilds per platform since the last one.

**Comment by FRidh** [comment]
Thanks. Note python-unstable builds the exact same revision as staging-next now.

**Comment by vcunat** [comment]
Oh, I looked wrong, I was one evaluation behind.  Anyway, only current version is being built now.

**Comment by vcunat** [comment]
@grahamc any idea how to get Hydra job-management rights to @FRidh?

**Comment by vcunat** [comment]
ghc on darwin shows a problem around sphinx, I suppose?
```
make -C utils/haddock/doc html SPHINX_BUILD=/nix/store/xd97694caj013pbf54mi3gnhk3jw5qga-python3.7-sphinx-1.8.3/bin/sphinx-build
/nix/store/xd97694caj013pbf54mi3gnhk3jw5qga-python3.7-sphinx-1.8.3/bin/sphinx-build -b html . .build-html
Traceback (most recent call last):
  File "/nix/store/xd97694caj013pbf54mi3gnhk3jw5qga-python3.7-sphinx-1.8.3/bin/.sphinx-build-wrapped", line 7, in <module>
    from sphinx.cmd.build import main
  File "/nix/store/xd97694caj013pbf54mi3gnhk3jw5qga-python3.7-sphinx-1.8.3/lib/python3.7/site-packages/sphinx/cmd/build.py", line 25, in <module>
    from sphinx.application import Sphinx
  File "/nix/store/xd97694caj013pbf54mi3gnhk3jw5qga-python3.7-sphinx-1.8.3/lib/python3.7/site-packages/sphinx/application.py", line 38, in <module>
    from sphinx.registry import SphinxComponentRegistry
  File "/nix/store/xd97694caj013pbf54mi3gnhk3jw5qga-python3.7-sphinx-1.8.3/lib/python3.7/site-packages/sphinx/registry.py", line 19, in <module>
    from pkg_resources import iter_entry_points
ModuleNotFoundError: No module named 'pkg_resources'
```

**Comment by FRidh** [comment]
That's a regression from f7e28bf5d8181926e600a222cb70180519d09726. Looking at the changeset `setuptools` is no longer propagated and thus no longer added to the wrapper/injected. While this is desirable long-term, I did not intend to do this right now because this will cause more run-time issues. I can push a fix directly but this will cause a whole new rebuild again.

**Comment by FRidh** [comment]
I've pushed a fix for specifically sphinx but not buildPython* in general. Depending on the amount of runtime failures we encounter we can always fix it there.

**Comment by vcunat** [comment]
Any idea whether this is still attempted for 19.09?

**Comment by FRidh** [comment]
@disassembler mentioned on Discourse the branch-off was postponed till tomorrow so we can get this in. I think later tonight we should simply merge, despite a chance in regressions.

**Comment by FRidh** [comment]
Jobs here can be cancelled https://hydra.nixos.org/jobset/nixpkgs/python-unstable

**Comment by vcunat** [comment]
Cancelled.  `staging-next` is showing ~1k regressions against "trunk" already, though much of that might be some random failures.  Perhaps it depends whether "we" specifically want this change in 19.09 (or more precisely, whether the release managers do).

**Comment by FRidh** [comment]
Qt 4 is broken on staging-next, that's causing the x86_64-linux failures. Aarch64 is also having more regressions.

**Comment by FRidh** [comment]
@vcunat this PR for Qt48 and gcc8 ? https://github.com/NixOS/nixpkgs/pull/67246

**Comment by vcunat** [comment]
That change was merged into gcc8 branch but I can't see it in staging-next :-/

**Comment by FRidh** [comment]
Maybe gcc8 got force-pushed removing it

**Comment by vcunat** [comment]
No, see comments on https://github.com/NixOS/nixpkgs/pull/63132

**Comment by FRidh** [comment]
@vcunat will you merge gcc-8 or shall I?

**Comment by vcunat** [comment]
I can't make it today.

**Comment by FRidh** [comment]
OK. Now gcc-8 is in (5f4734b1dd8410d5287fe0f22b9c0f30850a2fc3).

**Comment by andir** [comment]
Would you consider picking #68096 for this iteration of staging next as well? It seems to be fairly low impact on the overall systems besides the extra rebuilds. Having this through staging would help us having it accepted for 19.09.

**Comment by vcunat** [comment]
Oh, I wondered if this release should only get one large feature merged on the last day (which is past the _planned_ fork-off date), but apparently we end up the same on every release :-)

**Comment by FRidh** [comment]
proposed eval fix https://github.com/NixOS/nixpkgs/pull/68324

**Comment by nh2** [comment]
It seems this merge broke building my builds with musl (due to a new `[-Werror=attribute-alias]` error appearing when `glibc` is being built with musl):

https://github.com/nh2/static-haskell-nix/issues/56

**Comment by nh2** [comment]
> It seems this merge broke building my builds with musl

PR to fix it: #71480

File: pkgs/development/interpreters/python/mk-python-derivation.nix [hunk 1]
```diff
@@ -71,48 +92,61 @@ if disabled
...
 
   inherit strictDeps;
 
   LANG = "${if python.stdenv.isDarwin then "en_US" else "C"}.UTF-8";
 
   # Python packages don't have a checkPhase, only an installCheckPhase
   doCheck = false;
-  doInstallCheck = doCheck;
-  installCheckInputs = checkInputs;
+  doInstallCheck = attrs.doCheck or true;
```
Comment (louisblin) [inline pkgs/development/interpreters/python/mk-python-derivation.nix:139]: After that change, when setting `doCheck = false` you get `doInstallCheck = false or true = true` and tests will still run. As far as I can tell, the python docs still recommend using `doCheck` to control whether tests run, so this looks like a regression.

Comment (vcunat) [reply pkgs/development/interpreters/python/mk-python-derivation.nix:139]: Nitpick: Nix's "or" works for missing attributes, not for "falsy values".

**Pull request merged**